pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -453,6 +490,9 @@ public CompletableFuture<Void> closeAsync() {
             }
             state = State.Closing;
 
+            // forcefully delete heartbeat topic when close broker
+            deleteHeartbeatResource();
```

Comment:
there's a need to add a check that the `state` before setting to `State.Closing` was `State.Started` before running this method. the reason for this is that the brokerId() gets assigned only after `start()` has been called.